connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ConnectMetricsRegistry.java

```diff
@@ -230,13 +246,57 @@ public ConnectMetricsRegistry(Set<String> tags) {
                                                "committed/flushed/acknowledged by the sink task.",
                                                sinkTaskTags);
         sinkRecordActiveCountMax = createTemplate("sink-record-active-count-max", SINK_TASK_GROUP_NAME,
-                                                  "The maximum number of records that have been read from Kafka but not yet completely " +
-                                                  "committed/flushed/acknowledged by the sink task.",
+                                                  "The maximum number of records that have been read from Kafka but not yet completely "
+                                                  + "committed/flushed/acknowledged by the sink task.",
                                                   sinkTaskTags);
         sinkRecordActiveCountAvg = createTemplate("sink-record-active-count-avg", SINK_TASK_GROUP_NAME,
-                                                  "The average number of records that have been read from Kafka but not yet completely " +
-                                                  "committed/flushed/acknowledged by the sink task.",
+                                                  "The average number of records that have been read from Kafka but not yet completely "
+                                                  + "committed/flushed/acknowledged by the sink task.",
                                                   sinkTaskTags);
+
+        /***** Worker level *****/
+        Set<String> workerTags = new LinkedHashSet<>(tags);
+
+        connectorCount = createTemplate("connector-count", WORKER_GROUP_NAME, "The number of connectors run in this worker.", workerTags);
+        taskCount = createTemplate("task-count", WORKER_GROUP_NAME, "The number of tasks run in this worker.", workerTags);
+        workerStatus = createTemplate("status", WORKER_GROUP_NAME, "The state of this worker. One of: 'running', 'stopping', or 'stopped'.",
```

Comment:
Hmm, this seems different than the KIP and seems to provide less information, e.g. rebalancing never shows up. I figured this would either reuse the enum from AbstractCoordinator or put a new one in AbstractHerder that the herders then update. Despite the class name matching, I think this should really refer to the worker process rather than the state of the `Worker` class. The class is really just a container for connectors/tasks so it doesn't have enough information to understand the state of the whole process.

Reply:
The problem is that the worker doesn't really known when a rebalance is occurring -- the herders uses the worker, but the distributed worker doesn't really inform the worker when it is rebalancing. That's why I added the `rebalancingMode` to the rebalance metrics, which returns 1 when rebalancing or 0 when not. WDYT?

Reply:
ok, so if the info is available somewhere else that makes sense. then does this metric really add any value? it's only not `running` very early after the process starts or briefly when shutting down. I guess it kind of helps debug if you get stuck doing something at startup/shutdown, but is that useful as a metric given that you just have to look at logs anyway?

Reply:
Good question about whether it's valuable. About the only thing it might help identify is a worker process that has been told to stop but is blocked waiting for any connectors and/or tasks to stop (in [lines 166-172](https://github.com/apache/kafka/pull/4011/files#diff-316d2c222b623ee65e8065863bf4b9ceR166-172)).

Reply:
@ewencp okay, since the worker's `status` metric has virtually no value, it's better to not have it. Added a commit that removes it.